webClient/src/app/core/menu-bar/menu-bar.component.ts

```diff
@@ -107,15 +133,30 @@ export class MenuBarComponent implements OnInit {
     // this.editorControl.saveAllFile.subscribe(x => {
     //   this.saveAll();
     // });
+
   }
 
-  private resetLanguageSelectionMenu() {
-    const monaco = this.editorControl.editorCore.getValue();
-    if (!monaco) {
-      this.log.warn(`Language registered without monaco present`);
-      return;
+  public getMenuItemStyle(menuItem) {
+    let style = [];
+    if (menuItem.name === 'group-end') {
+      style.push('group-line');
+    }
+    const editor = this.editorControl.editor.getValue();
+    if (editor) {
+      if (menuItem.isDisabled
+          && menuItem.isDisabled({
+            editor: editor,
+            controller: this.editorControl,
+            log: this.log
+          })) {
+        style.push('disabled');
+      }
     }
-    this.languageSelectionMenu.children = monaco.languages.getLanguages().sort(function(lang1, lang2) {
+    return style;
+  }
+
+  private resetLanguageSelectionMenu() {
+    this.languageSelectionMenu.children = this.monaco.languages.getLanguages().sort(function(lang1, lang2) {
```

Comment:
I know you're creating your own monaco object, which is good for potential future versions of Editor where monaco isn't a global, but shouldn't we still check if it's set to something? Wouldn't it still throw the error if its not? Or does no such case exist, and if that's true, why did we even have that log statement there to begin with

Reply:
Could have been more careful but didn't bother due to timing: it better exist at this point or something's very wrong.
I'm assuming editorCore is loaded before languages have the opportunity to be reset. This is correct at the moment and if it ever changed it would be weird.